tests/conftest.py

```diff
@@ -325,3 +326,19 @@ def celery_app(app):
 def treestatus_url():
     """A string holding the Tree Status base URL."""
     return "http://treestatus.test"
+
+
+def pytest_assertrepr_compare(op, left, right):
+    if isinstance(left, JSONResponse) and isinstance(right, int) and op == "==":
+        # Hook failures when comparing JSONResponse objects so we get the detailed
+        # failure description from inside the response object contents.
+        #
+        # The following example code would trigger this hook:
+        #
+        #   response = client.get()
+        #   assert response == 200  # Fails if response is HTTP 401, triggers this hook
```

Comment:
Why are you comparing the response object itself to the status code? I'm not sure what implementing this hook is providing to us here?

Reply:
There is also a custom assertion reporter in pytest-flask that closely resembles what I wrote here.  My version includes the JSON response body where the pytest-flask reporter provides just the status code and status name.